### DISCUSSION TITLE
Weighted deployments?

### DISCUSSION THREAD

**Discussion by thansk** [caprover/caprover#2018] [Q&A]
I would like to redirect 10-25% of users to the new deployment and keep the rest on the old one as a safeguard until I am sure everything is good. 

Is that doable with CapRover?

**Comment by githubsaturn** [comment]
CapRover doesn't have any special logic here. It uses Swarm's rolling updates. You can update these parameters by doing something like: 
```
docker service update srv-captain-myapp --update-*****
```
For example:
```
docker service update --update-parallelism 1 --update-delay 30s srv-captain-myapp 
```

See https://docs.docker.com/reference/cli/docker/service/update/

**Reply by thansk** [reply]
So this command (but with `--update-delay 1h`) would set the setting for all future deployments, and if I have, for example, some full stack application with 4 instances, the first hour, 25% of the users will get the new version, the second hour - 50% and so on?

How is traffic split between the instances? Is it per request? If so, traffic will get mixed and make it pointless to detect if there are any bugs, it might even create more.

**Reply by githubsaturn** [reply]
Yes and yes. The traffic is being handled by docker swarm, and that uses a simple round robin load distribution.
